File: app/src/main/java/com/glucocontrol/data/export/CsvExporter.kt
package com.glucocontrol.data.export

import com.glucocontrol.domain.model.GlucoseReading
import java.io.OutputStream
import javax.inject.Inject
import javax.inject.Singleton

// El separador es coma; las notas se envuelven entre comillas y las comillas
// internas se duplican según RFC 4180.
@Singleton
class CsvExporter @Inject constructor() {
    fun export(readings: List<GlucoseReading>, stream: OutputStream) {
        stream.bufferedWriter(Charsets.UTF_8).use { writer ->
            writer.write("id,fecha,hora,glucosa_mg_dl,etiqueta,notas\n")
            readings.forEach { r ->
                val hora = r.time?.toString() ?: ""
                val notas = r.notes
                    ?.replace("\"", "\"\"") // escapar comillas internas
                    ?.let { "\"$it\"" }
                    ?: ""
                writer.write("${r.id},${r.date},$hora,${r.valueMgDl},${r.tag.name},$notas\n")
            }
        }
    }
}

File: lib/data/export/csv_exporter.dart
import 'package:csv/csv.dart';
import 'package:glucocontrol/domain/model/glucose_reading.dart';
import 'package:intl/intl.dart';

/// Genera un archivo CSV RFC 4180 con las lecturas indicadas.
class CsvExporter {
  static const _headers = [
    'Fecha',
    'Hora',
    'Glucosa (mg/dL)',
    'Etiqueta',
    'Notas',
  ];

  String export(List<GlucoseReading> readings) {
    final dateFmt = DateFormat('dd/MM/yyyy', 'es');
    final timeFmt = DateFormat('HH:mm', 'es');

    final rows = <List<dynamic>>[_headers];

    for (final r in readings) {
      rows.add([
        dateFmt.format(r.date),
        r.time != null ? timeFmt.format(r.time!) : '',
        r.valueMgDl,
        r.tag.label,
        r.notes ?? '',
      ]);
    }

    return const ListToCsvConverter().convert(rows);
  }
}
